letsencrypt/client/tests/revoker_test.py

```diff
@@ -0,0 +1,390 @@
+"""Test :mod:`letsencrypt.client.revoker`."""
+import csv
+import os
+import pkg_resources
+import shutil
+import tempfile
+import unittest
+
+import mock
+
+from letsencrypt.client import errors
+from letsencrypt.client import le_util
+
+
+class RevokerBase(unittest.TestCase):  # pylint: disable=too-few-public-methods
+    """Base Class for Revoker Tests."""
+    def setUp(self):
+        self.paths, self.certs, self.key_path = create_revoker_certs()
+
+        self.backup_dir = tempfile.mkdtemp("cert_backup")
+        self.mock_config = mock.MagicMock(cert_key_backup=self.backup_dir)
+
+        self.list_path = os.path.join(self.backup_dir, "LIST")
+
+    def _store_certs(self):
+        # pylint: disable=protected-access
+        from letsencrypt.client.revoker import Revoker
+        Revoker.store_cert_key(self.paths[0], self.key_path, self.mock_config)
+        Revoker.store_cert_key(self.paths[1], self.key_path, self.mock_config)
+
+        # Set metadata
+        for i in xrange(2):
+            self.certs[i].add_meta(
+                i, self.paths[i], self.key_path,
+                Revoker._get_backup(self.backup_dir, i, self.paths[i]),
+                Revoker._get_backup(self.backup_dir, i, self.key_path))
+
+    def _get_rows(self):
+        with open(self.list_path, "rb") as csvfile:
+            return [row for row in csv.reader(csvfile)]
+
+    def _write_rows(self, rows):
+        with open(self.list_path, "wb") as csvfile:
+            csvwriter = csv.writer(csvfile)
+            for row in rows:
+                csvwriter.writerow(row)
+
+
+class RevokerTest(RevokerBase):
+    def setUp(self):
+        from letsencrypt.client.revoker import Revoker
+        super(RevokerTest, self).setUp()
+
+        with open(self.key_path) as key_file:
+            self.key = le_util.Key(self.key_path, key_file.read())
+
+        self._store_certs()
+
+        self.mock_installer = mock.MagicMock()
+        self.revoker = Revoker(self.mock_installer, self.mock_config)
+
+    def tearDown(self):
+        shutil.rmtree(self.backup_dir)
```

Comment:
btw, py.test has a nice mechanism to require, create and clean up such temp dirs.
you basically just say you need a tempdir, that is all. rest is done automagically.
just saying. ;)